src/components/Welcome.js

```diff
@@ -1,11 +1,17 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 
-const Welcome = () => {
+const Welcome = ({ onClick }) => {
   return (
     <div>
       <h1>Welcome</h1>
+      <button onClick={onClick}>Create a New List</button>
     </div>
   );
 };
+/* Checks that onClick is a function and is required as a prop */
+Welcome.propTypes = {
+  onClick: PropTypes.func.isRequired,
```

Comment:
I've never used PropTypes before, is this like a type checker?

Reply:
PropTypes is kind of like a type-checker, yeah. The only difference is that PropTypes checks for the types at _run time_ while things like TypeScript or Flow check them at _build time_. If you use the wrong type of prop, PropTypes will log a warning in your console while the app is running, but it will still "work" (as long as the prop type mismatch doesn't break your code!). TypeScript or Flow would actually prevent you from deploying code which contained such a bug in the first place. 
It's a good incremental step in this direction. PropTypes is a great way to add this kind of thing without worrying about additional tooling or type syntax. 
More docs and info: https://reactjs.org/docs/typechecking-with-proptypes.html